JWT token with HS256 and corresponding JWK file : UnresolvableKeyException

### Describe the bug

My quarkus backend receives requests with JWT Token with some information.
I tried with quarkus-smallrye-jwt, and it worked for RS256 algorithm. I created a token with jwt.io and sent it with my request in my test, it has been correctly decoded.

To verify I simply used :
`@Inject
    JsonWebToken jwt;
  ...
  @GET
    public Response get() {
        System.out.println(hasJwt());
        System.out.println(jwt.getClaimNames());
`

I tried with pem file and jwk file then in smallrye.jwt.verify.key.location, and it was OK.
Then we moved to HS256 algorithm because of our needs, I created corresponding jwk file and token but this time it has not been working...

### Expected behavior

I want to be able to simply use HS256 algorithm for encoding/decoding JWT Tokens with quarkus-smallrye-jwt

### Actual behavior

I created JWKS file for HS256 algorithm with this format :
`{
    "keys": [
        {
            "kty":"oct",
            "k":"uWlwBLGv4EpifZ52EhTuU9L-76AF9Vf4yumSD1P-2uE",
            "alg":"HS256"
        }
    ]
}`

I encoded a JWT token with HS256 and same secret, and sent it to quarkus back but I got this error :

2022-05-17 21:15:09,295 DEBUG [io.qua.sma.jwt.run.aut.MpJwtValidator] (executor-thread-0) Authentication failed: io.smallrye.jwt.auth.principal.ParseException: SRJWT07000: Failed to verify a token
	at io.smallrye.jwt.auth.principal.DefaultJWTTokenParser.parseClaims(DefaultJWTTokenParser.java:161)
	at io.smallrye.jwt.auth.principal.DefaultJWTTokenParser.parse(DefaultJWTTokenParser.java:56)
	at io.smallrye.jwt.auth.principal.DefaultJWTCallerPrincipalFactory.parse(DefaultJWTCallerPrincipalFactory.java:31)
	at io.smallrye.jwt.auth.principal.DefaultJWTParser.parse(DefaultJWTParser.java:64)
	at io.smallrye.jwt.auth.principal.DefaultJWTParser_ClientProxy.parse(Unknown Source)
        ...
Caused by: org.jose4j.lang.UnresolvableKeyException: SRJWT07004: Failed to load a key from the key content while resolving
	at io.smallrye.jwt.auth.principal.AbstractKeyLocationResolver.reportUnresolvableKeyException(AbstractKeyLocationResolver.java:187)
	at io.smallrye.jwt.auth.principal.KeyLocationResolver.resolveKey(KeyLocationResolver.java:66)
	at org.jose4j.jwt.consumer.JwtConsumer.processContext(JwtConsumer.java:213)
	at org.jose4j.jwt.consumer.JwtConsumer.process(JwtConsumer.java:426)
	at io.smallrye.jwt.auth.principal.DefaultJWTTokenParser.parseClaims(DefaultJWTTokenParser.java:143)
	... 74 more


### How to Reproduce?

Add quarkus-smallrye-jwt dependency, create JWK file for HS256 algorithm, add it to smallrye.jwt.verify.key.location. Encode JWT token with same secret, with jwt.io for example, and add it in "Authorization" header of request for quarkus backend

### Output of `uname -a` or `ver`

Darwin 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:45:05 PDT 2022; root:xnu-8020.101.4~15/RELEASE_X86_64 x86_64

### Output of `java -version`

OpenJDK 64-Bit Server VM AdoptOpenJDK (build 14.0.2+12, mixed mode, sharing)

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.8.1.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/SignSecretKeyUnitTest.java
+++ b/extensions/smallrye-jwt/deployment/src/test/java/io/quarkus/jwt/test/SignSecretKeyUnitTest.java
@@ -1,0 +1,34 @@
+package io.quarkus.jwt.test;
+
+import static org.hamcrest.Matchers.equalTo;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.restassured.RestAssured;
+import io.smallrye.jwt.build.Jwt;
+
+public class SignSecretKeyUnitTest {
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .withApplicationRoot((jar) -> jar
+                    .addClass(DefaultGroupsEndpoint.class)
+                    .addAsResource("secretKey.jwk")
+                    .addAsResource("applicationSignSecretKey.properties", "application.properties"));
+
+    /**
+     * Validate a request with MP-JWT without a 'groups' claim is successful
+     * due to the default value being provided in the configuration
+     *
+     */
+    @Test
+    public void echoGroups() {
+        String token = Jwt.upn("upn").groups("User").sign();
+        RestAssured.given().auth()
+                .oauth2(token)
+                .get("/endp/echo")
+                .then().assertThat().statusCode(200)
+                .body(equalTo("User"));
+    }
+}

--- a/extensions/smallrye-jwt/deployment/src/test/resources/applicationSignSecretKey.properties
+++ b/extensions/smallrye-jwt/deployment/src/test/resources/applicationSignSecretKey.properties
@@ -1,0 +1,8 @@
+smallrye.jwt.verify.key.location=/secretKey.jwk
+smallrye.jwt.sign.key.location=/secretKey.jwk
+smallrye.jwt.new-token.issuer=https://server.example.com
+mp.jwt.verify.issuer=https://server.example.com
+smallrye.jwt.verify.algorithm=HS256
+
+quarkus.log.category."io.quarkus.smallrye.jwt.runtime.auth.MpJwtValidator".min-level=TRACE
+quarkus.log.category."io.quarkus.smallrye.jwt.runtime.auth.MpJwtValidator".level=TRACE

--- a/extensions/smallrye-jwt/deployment/src/test/resources/secretKey.jwk
+++ b/extensions/smallrye-jwt/deployment/src/test/resources/secretKey.jwk
@@ -1,0 +1,6 @@
+{
+ "keys":
+ [
+  { "kty":"oct", "k":"uWlwBLGv4EpifZ52EhTuU9L-76AF9Vf4yumSD1P-2uE", "alg":"HS256" }
+ ]
+}

